### PR TITLE
Improve current operation display format

### PR DESCRIPTION
**Pull request by Dezzy12**
Hello, I implemented a fix for issue: #12 

Updated string_of_operation function in interactive_viewer.ml to provide a clearer and more explicit representation of the current operation. This change will enhance readability and simplify the display format. 
Please review!

### CODE REVIEW

File: lib/interactive_viewer.ml [hunk 1]
```diff
@@ -9,7 +9,16 @@ let current_patch patches =
   let$ index = index in
   List.nth_opt patches index
 
-let string_of_operation = Format.asprintf "%a" (Patch.pp_operation ~git:false)
+let string_of_operation operation =
+  match operation with
+  | Patch.Create path -> "Creation of `" ^ path ^ "`"
```
Comment (Julow) [inline lib/interactive_viewer.ml:14]: The backquotes `` ` .. ` `` don't look very nice, we could use some colors or maybe bold ?

This function is called below like this:
```ocaml
  W.string @@ string_of_operation p.Patch.operation
```
`W.string` has a `~attr` optional argument that can be used to put some text in colors/bold/etc..
See the doc for `W.string`: https://ocaml.org/p/nottui/latest/doc/Nottui_widgets/index.html#val-string

We can then use `Ui.hcat` ([doc](https://ocaml.org/p/nottui/latest/doc/Nottui/Ui/index.html#val-hcat)) to concatenate a list of elements created with `W.string`.

**Review by Julow — COMMENTED** [review]
Looks very nice :)

I've got an idea to improve the output by using attributes. What do you think?

File: lib/interactive_viewer.ml [hunk 2]
```diff
@@ -9,7 +9,16 @@ let current_patch patches =
...
+  | Patch.Rename (old_path, new_path) ->
+      "Rename from `" ^ old_path ^ "` to `" ^ new_path ^ "`"
+  | Patch.Delete path -> "Deletion of `" ^ path ^ "`"
+  | Patch.Edit path -> "Modification of `" ^ path ^ "`"
+  | Patch.Rename_only (old_path, new_path) ->
+    "Rename only from `" ^ old_path ^ "` to `" ^ new_path ^ "`"
```
[the remark this answers is not in the document]
Comment (Dezzy12) [reply lib/interactive_viewer.ml:20]: Oh yeah that is right.

**Comment by Dezzy12** [comment]
@Julow I have implemented suggested changes

**Review by panglesd — APPROVED** [review]
Looks good!

An improvement I can think of is to factor all those attributes in three variables, and use those variables.

File: lib/interactive_viewer.ml [hunk 3]
```diff
@@ -9,13 +9,44 @@ let current_patch patches =
...
+    Ui.hcat [
+      W.string "Rename ";
+      W.string ~attr:Notty.A.(fg blue ++ st bold) ("" ^ old_path ^ "");
+      W.string " to ";
+      W.string ~attr:Notty.A.(fg green ++ st bold) ("" ^ new_path ^ "");
+    ]
+  | Patch.Edit path ->
+    Ui.hcat [
+      W.string "Modification of ";
+      W.string ~attr:Notty.A.(fg red ++ st bold) ("" ^ path ^ "")
```
Comment (panglesd) [inline lib/interactive_viewer.ml:41]: I would have chosen `blue` for the edit color: `red` makes it seem to have been deleted...

**Comment by Dezzy12** [comment]
Thanks @panglesd, Changes implemented

File: lib/interactive_viewer.ml [hunk 4]
```diff
@@ -9,13 +9,48 @@ let current_patch patches =
   let$ index = index in
   List.nth_opt patches index
 
-let string_of_operation = Format.asprintf "%a" (Patch.pp_operation ~git:false)
+let greenBoldAttr = Notty.A.(fg green ++ st bold)
+let redBoldAttr = Notty.A.(fg red ++ st bold)
+let blueBoldAttr = Notty.A.(fg blue ++ st bold)
+
+let string_of_operation operation =
```
[the remark this answers is not in the document]
Comment (Dezzy12) [reply lib/interactive_viewer.ml:16]: I think a function name like `render_operation` will do. What do you think?

[the remark this answers is not in the document]
Comment (panglesd) [reply lib/interactive_viewer.ml:16]: That, or `ui_of_operation` to keep it the same general scheme!

**Comment by Dezzy12** [comment]
I have implemented the changes

**Review by panglesd — APPROVED** [review]

**Comment by panglesd** [comment]
Thanks @Dezzy12 for your nice contribution!

**Comment by Julow** [comment]
@Dezzy12 Congrat for the fantastic work! We need a diff file that contains all the operations to properly test this feature. What do you think of solving this issue ? https://github.com/panglesd/diffcessible/issues/28

**Comment by Dezzy12** [comment]
Thank you @Julow 
And yes please assign the issue: #28

**Pull request merged**